code-server/README.md

```diff
@@ -79,3 +79,30 @@ module "code-server" {
   extensions = ["dracula-theme.theme-dracula", "ms-azuretools.vscode-docker"]
 }
 ```
+
+### Offline and Use Cached Modes
+
+By default the module looks for code-server at `/tmp/code-server` but this can be changed with `install_prefix`.
+
+Run an existing copy of code-server if found, otherwise download from GitHub:
+
+```tf
+module "code-server" {
+  source     = "registry.coder.com/modules/code-server/coder"
+  version    = "1.0.8"
+  agent_id   = coder_agent.example.id
+  use_cached = true
+  extensions = ["dracula-theme.theme-dracula", "ms-azuretools.vscode-docker"]
+}
+```
+
+Just run code-server in the background, don't fetch it from GitHub:
+
+```tf
+module "code-server" {
+  source   = "registry.coder.com/modules/code-server/coder"
```

Comment:
`offline` mode will work great if `code-server` is prebuilt into the image. However, for offline deployments, they may not have access to registry.coder.com to fetch the module and can not use the feature unless they do something like [this](https://coder.com/docs/v2/latest/templates/modules#offline-installations).

Reply:
@matifali  - true and for our use case we would use a [private module registry](https://developer.hashicorp.com/terraform/cloud-docs/registry/publish-modules) and for the running `code-server` instance, it would use the [code-marketplace](https://github.com/coder/code-marketplace).

But not your policy for the control plane running terraform could have egress, but the running workspace won't

Reply:
@matifali  - Offline docs does cover alot of this, and companies might have flexibility to access `registry.terraform.io` but not github sh installs.

https://coder.com/docs/v2/latest/install/offline#offline-deployments

Reply:
Terraform Enterprise also have good docs on how side load providers into an air gapped install of terraform enterprise. 

https://developer.hashicorp.com/terraform/enterprise/registry/airgapped-providers

Reply:
Terraform Enterprise also have good docs on how side load providers into an air gapped install of terraform enterprise. 

https://developer.hashicorp.com/terraform/enterprise/registry/airgapped-providers

Reply:
Thanks for the great PR and the explanations. 🎉